packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -856,11 +861,39 @@ export function TransferPanel() {
         )
       }
 
+      // when sending additional ETH with ERC-20, we add the additional ETH value as maxSubmissionCost
+      let maxSubmissionCostOverride: BigNumber | undefined
+      const isValidBatchTransfer =
+        isBatchTransferSupported && Number(amount2) > 0
+
+      if (isValidBatchTransfer) {
+        const gasEstimates = (await bridgeTransferStarter.transferEstimateGas({
+          amount: amountBigNumber,
+          signer
+        })) as DepositGasEstimates
+
+        if (!gasEstimates.estimatedChildChainSubmissionCost) {
+          throw 'Failed to estimate deposit maxSubmissionCost'
+        }
+
+        maxSubmissionCostOverride = utils
+          .parseEther(amount2)
+          .add(gasEstimates.estimatedChildChainSubmissionCost)
+      }
+
       // finally, call the transfer function
       const transfer = await bridgeTransferStarter.transfer({
         amount: amountBigNumber,
         signer,
-        destinationAddress
+        destinationAddress,
+        overrides: {
+          maxSubmissionCost: isValidBatchTransfer
+            ? maxSubmissionCostOverride
+            : undefined,
+          excessFeeRefundAddress: isValidBatchTransfer
+            ? destinationAddress
+            : undefined
+        }
```

Comment:
I prefer something like this

the text on the error toast is just a draft to demo

```typescript
      // when sending additional ETH with ERC-20, we add the additional ETH value as maxSubmissionCost
      const isValidBatchTransfer =
        isBatchTransferSupported && Number(amount2) > 0

      const gasEstimates = (await bridgeTransferStarter.transferEstimateGas({
        amount: amountBigNumber,
        signer
      })) as DepositGasEstimates

      if (
        isValidBatchTransfer &&
        !gasEstimates.estimatedChildChainSubmissionCost
      ) {
        errorToast(
          'Failed to estimate deposit maxSubmissionCost, ETH will not be transferred along with the token.'
        )
        throw 'Failed to estimate deposit maxSubmissionCost'
      }

      const overrides = isValidBatchTransfer
        ? {
            maxSubmissionCost: utils
              .parseEther(amount2)
              .add(gasEstimates.estimatedChildChainSubmissionCost),
            excessFeeRefundAddress: destinationAddress
          }
        : undefined
```